src/components/Products/SectionProducts.tsx

```diff
@@ -0,0 +1,89 @@
+import { useEffect, useState } from "preact/hooks"
+import { pick } from "@nosto/search-js/utils"
+import { useDecoratedSearchResults, useNostoAppState } from "@nosto/search-js/preact/hooks"
+import style from "./Products.module.css"
+import { cl } from "@nosto/search-js/utils"
+import { hitDecorators, defaultSize } from "@/config"
+
+export interface SectionProductsProps {
+  sectionId: string
+}
+
+interface BatchProps {
+  sectionId: string
+  handles: string[]
+}
+
+function Batch({ sectionId, handles }: BatchProps) {
+  const [html, setHtml] = useState<string>("")
+  const [loading, setLoading] = useState<boolean>(false)
+  const [error, setError] = useState<string | null>(null)
+
+  useEffect(() => {
```

Comment:
replace with new hook called useFetch(url) that should be used like this:

`const { html, loading, error } = useFetch(url)`